servers/src/main/java/tachyon/web/WorkerUIWebServer.java

```diff
@@ -22,25 +22,26 @@
 import com.google.common.base.Preconditions;
 
 import tachyon.conf.TachyonConf;
-import tachyon.worker.block.BlockWorker;
+import tachyon.worker.block.BlockDataManager;
 
 /**
  * A worker's UI web server
  */
 public class WorkerUIWebServer extends UIWebServer {
 
-  public WorkerUIWebServer(String serverName, InetSocketAddress address,
-      BlockWorker worker, TachyonConf conf) {
-    super(serverName, address, conf);
-    Preconditions.checkNotNull(worker, "WorkerStorage cannot be null");
-
-    mWebAppContext.addServlet(new ServletHolder(
-        new WebInterfaceWorkerGeneralServlet(worker)), "/home");
-    mWebAppContext.addServlet(new ServletHolder(
-        new WebInterfaceWorkerBlockInfoServlet(worker, conf)), "/blockInfo");
-    mWebAppContext.addServlet(new ServletHolder(
-        new WebInterfaceDownloadLocalServlet()), "/downloadLocal");
-    mWebAppContext.addServlet(new ServletHolder(
-        new WebInterfaceBrowseLogsServlet(false)), "/browseLogs");
+  public WorkerUIWebServer(String serverName, InetSocketAddress webAddress,
+      BlockDataManager blockdataManager, InetSocketAddress workerAddress, long startTimeMs,
+      TachyonConf conf) {
+    super(serverName, webAddress, conf);
+    Preconditions.checkNotNull(blockdataManager, "Block data manager cannot be null");
```

Comment:
Yes, will check the workerAddress as well. Others are checked by the base class.